### PR TITLE
(PUP-7732) Audit Acceptance Tests

### PR DESCRIPTION
**Pull request by justinstoller**
This audits the second half of the puppet acceptance tests. It is a companion PR to https://github.com/puppetlabs/puppet/pull/6040  I will need to use the PUP ticket prior to merging.

### CODE REVIEW

**Comment by justinstoller** [comment]
ping @johnduarte

File: acceptance/tests/ticket_2455_on_solaris_init_provider_should_start_service_in_own_smf_contract.rb [hunk 1]
```diff
@@ -1,5 +1,9 @@
 test_name "(PUP-2455) Service provider should start Solaris init service in its own SMF contract"
 
+tag 'audit:medium',
+    'audit:refactor',   # Use block style `test_name`
```
Comment (johnduarte) [inline acceptance/tests/ticket_2455_on_solaris_init_provider_should_start_service_in_own_smf_contract.rb]: use mk_tmp_environment_with_teardown

File: acceptance/tests/ticket_2455_on_solaris_init_provider_should_start_service_in_own_smf_contract.rb [hunk 2]
```diff
@@ -1,5 +1,9 @@
...
+    'audit:integration' # Service provider functionality
```
Comment (johnduarte) [inline acceptance/tests/ticket_2455_on_solaris_init_provider_should_start_service_in_own_smf_contract.rb]: acceptance? this is a solaris smf specific issue.

File: acceptance/tests/ticket_4289_facter_should_recognize_OEL_operatingsystemrelease.rb [hunk 3]
```diff
@@ -7,6 +7,11 @@
 
 test_name "#4289: facter should recognize OEL operatingsystemrelease"
 
+tag 'audit:low',     # Special Facter OEL handling
+    'audit:refactor', # Use block style `test_name`
+    'audit:unit'     # This is OS specific, but I would assume that
+                     # facter unit tests would catch this...
+
```
Comment (johnduarte) [inline acceptance/tests/ticket_4289_facter_should_recognize_OEL_operatingsystemrelease.rb]: 'audit:delete'  # this is a facter test

File: acceptance/tests/windows/env_windows_installdir_fact.rb [hunk 4]
```diff
@@ -3,6 +3,9 @@
 # present and accurate.
 test_name 'PA-466: Ensure env_windows_installdir fact is present and correct' do
 
+  tag 'audit:low',       # important runtime environment/packaging integration, rarely changed?
+      'audit:acceptance'
```
Comment (johnduarte) [inline acceptance/tests/windows/env_windows_installdir_fact.rb]: 'audit:delete'  # move test to puppet-agent suite

File: acceptance/tests/reports/corrective_change_via_puppet.rb [hunk 5]
```diff
@@ -4,7 +4,9 @@
 
 test_name "C98094 - a resource changed via Puppet manifest will not be reported as a corrective change" do
 
-  tag 'broken:images'
+  tag 'audit:medium',
+      'audit:integration',
+      'audit:refactor',    # Uses a server currently, but is testing agent report
```
Comment (justinstoller) [inline acceptance/tests/reports/corrective_change_via_puppet.rb]: d'oh

**Comment by justinstoller** [comment]
I'd would also like to get a run of the acceptance tests done locally before this is merged.

**Comment by johnduarte** [comment]
@justinstoller disappointed you didn't challenge some of my suggestions. :)

**Comment by justinstoller** [comment]
Watch what you wish for (;

**Comment by justinstoller** [comment]
So running this PR I get:
```
              Attempted: 339
                 Passed: 271
                 Failed: 6
                Errored: 8
                Skipped: 53
                Pending: 1
                  Total: 339
```
But I get those same results from running the current head of master: b76616f9ba9794d47bd986979e62dcfbc24eba17 ... So, I guess this is fine?

**Comment by johnduarte** [comment]
@justinstoller this PR should be targeted at the `acceptance-audit` branch. I changed the base which brought in a couple of extraneous commits. Can you rebase?

**Comment by justinstoller** [comment]
rebased

**Pull request merged**